### PR TITLE
Add new pipeline tasks

### PR DESCRIPTION
**Pull request by s-block**

### CODE REVIEW

File: image-build-to-ecr.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+platform: linux
+
+image_resource:
+  type: docker-image
+  source:
+    repository: concourse/docker-image-resource
+    tag: 'pr-227'
```
Comment (r4vi) [inline image-build-to-ecr.yaml]: hasn't this PR been merged ?

File: extract-policies-from-repos.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+platform: linux
+image_resource:
+  type: docker-image
+  source:
+    repository: quay.io/s_block/alpine-git-jq
```
Comment (r4vi) [inline extract-policies-from-repos.yaml]: move to MOJ?

**Review by r4vi — CHANGES_REQUESTED** [review]
can you move that repo to MOJ quay repo?

File: image-build-to-ecr.yaml (hunk 1 above)
Comment (s-block) [reply image-build-to-ecr.yaml]: Not sure, was from an existing task but will check it out

File: extract-policies-from-repos.yaml (hunk 2 above)
Comment (s-block) [reply extract-policies-from-repos.yaml]: Cool, has python as well now so will rename as well

File: diff-iam-policy.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+params:
+  AWS_ACCESS_KEY_ID: ((secrets.role-putter-access-key-id))
+  AWS_SECRET_ACCESS_KEY: ((secrets.role-putter-secret-access-key))
+inputs:
+- name: policies
+run:
+  path: sh
+  args:
+  - -ec
+  - |
```
Comment (andyhd) [inline diff-iam-policy.yaml]: Not finished? Need to remove the script from the YAML and run the separated script instead

File: image-build-to-ecr.yaml (hunk 1 above)
Comment (s-block) [reply image-build-to-ecr.yaml]: Yep, removed tag

File: extract-policies-from-repos.yaml (hunk 2 above)
Comment (s-block) [reply extract-policies-from-repos.yaml]: Moved to moj

File: diff-iam-policy.yaml (hunk 3 above)
Comment (s-block) [reply diff-iam-policy.yaml]: Forgot to commit yesterday. Have now changed to run scripts

**Review by r4vi — APPROVED** [review]

File: extract-policies-from-repos.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  GITHUB_TOKEN: ((secrets.github-access-token))
+inputs:
+- name: repos
+- name: common-tasks
+outputs:
+- name: policies
+run:
+  path: bash
+  args:
+  - ./common-tasks/scripts/extract-policies-from-repos.sh
```
Comment (andyhd) [inline extract-policies-from-repos.yaml:17]: If you make the script executable, you could have
```
run:
  path: ./common-tasks/scripts/extract-policies-from-repos.sh
```

File: scripts/diff-iam-policy.sh [hunk 5]
```diff
@@ -0,0 +1,18 @@
+set -e
+
+cat policies/list.json | jq -r '.[] | [.app_name, .role, (.policy|tostring)] | @tsv' |
+  while IFS=$'\t' read -r app_name role policy; do
+    echo ${app_name}
+    echo ${role}
+    current=$(aws iam get-role-policy --role-name ${role} --policy-name ${app_name} | jq -r '.PolicyDocument')
+    echo ${current} | jq -S . > current.json
+    echo ${policy} | jq -S . > new.json
+    export ARE_SAME=$(jq --argfile a new.json --argfile b current.json -n '$a==$b')
```
Comment (andyhd) [inline scripts/diff-iam-policy.sh:10]: Do you need to `export` this variable?

File: scripts/diff-iam-policy.sh [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    if [[ \"${ARE_SAME}\" != \"true\" ]]; then
+      echo -e '\033[0;31mChange in iam_policy.json'
+      diff current.json new.json
+      exit 166
```
Comment (andyhd) [inline scripts/diff-iam-policy.sh:14]: You should probably explain this magic number, either with a comment or put it in a well-named variable

File: scripts/extract-policies-from-repos.sh [hunk 7]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+set -e
+
+pip install iam_builder
+echo '[]' | jq -S . > policies/list.json
+cat repos/list.txt
```
Comment (andyhd) [inline scripts/extract-policies-from-repos.sh:7]: This line seems unnecessary

**Pull request merged**